File: 2024/go/utils/file_utils.go
package utils

import (
	"bufio"
	"fmt"
	"os"
	"path/filepath"
)

func DataFileReader(date string, part string, verbose bool) ([]string, error) {

	currentDir, err := os.Getwd()

	if err != nil {
		fmt.Println("Error getting current directory:", err)
		return nil, err
	}
	if verbose {
		fmt.Println("Current directory:", currentDir)
	}

	parentDir := filepath.Dir(currentDir)

	if verbose {
		fmt.Println("Parent directory:", parentDir)
	}

	fullpath := parentDir + "/common/data/" + date + "-part" + part + "-test-data.txt"

	if verbose {
		fmt.Println("Full path:", fullpath)
	}

	file, err := os.Open(fullpath)

	if err != nil {
		return nil, err
	}
	defer file.Close()

	var lines []string
	scanner := bufio.NewScanner(file)

	for scanner.Scan() {
		lines = append(lines, scanner.Text())
	}

	if err := scanner.Err(); err != nil {
		return nil, err
	}

	return lines, nil
}

File: 2024/python/utils/file_utils.py
import os



def data_file_reader(filename = '', date = '', use_test_data = False, part = 1, verbose = False):
	current_directory = os.getcwd()
	parent_directory = os.path.dirname(current_directory)
	if verbose:
		print(f"current_directory: {current_directory}")
		print(f"parent_directory: {parent_directory}")

	filepath = f"{parent_directory}/common/data/{date}-{'part' + str(part) + '-' if use_test_data else ''}{'test-' if use_test_data else ''}data.txt"

	if filename:
		filepath = filename

	if verbose:
		print(f"filepath: {filepath}")

	with open(filepath, 'r') as file:
		lines = file.readlines()
	return [line.strip() for line in lines]
